### PR TITLE
Fix UnionArray is_null

### LINKED ISSUE #1625: UnionArray::is_null incorrect
**Describe the bug**

UnionArray::is_null checks the Union's validity map rather than its slot. 

Here is the default impl for `is_null`
https://github.com/apache/arrow-rs/blob/master/arrow/src/array/array.rs#L164-L166

There is no override in the union's impl of `Array`:https://github.com/apache/arrow-rs/blob/master/arrow/src/array/array_union.rs#L305-L313

See discussion from @tustvold and @viirya here:  https://github.com/apache/arrow-rs/pull/1589/files#r854817015

**To Reproduce**

**Expected behavior**
UnionArray::is_null should check the slot (child's data) for nullness, rather than its own

**Additional context**

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #1625.

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow/src/array/array_union.rs [hunk 1]
```diff
@@ -304,6 +304,24 @@ impl Array for UnionArray {
     fn data(&self) -> &ArrayData {
         &self.data
     }
+
```
Comment (tustvold) [inline arrow/src/array/array_union.rs:307]: Just to check, these special cases aren't necessary for correctness, but just clarity?

**Review by tustvold — APPROVED** [review]
LGTM :+1:

File: arrow/src/array/array_union.rs (hunk 1 above)
Comment (viirya) [reply arrow/src/array/array_union.rs:307]: I think so. `ArrayData.is_null` returns false if its `null_bitmap` is `None`. These values are not changed before/after this change.

**Comment by viirya** [comment]
Thank you @tustvold !

**Pull request merged**